src/Domain/WeightedElementDistribution.hpp

```diff
@@ -0,0 +1,147 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+template <size_t Dim>
+class Block;
+
+template <size_t Dim>
+class ElementId;
+
+namespace Spectral {
+enum class Quadrature;
+}  // namespace Spectral
+
+namespace domain {
+
+/*!
+ * \brief Distribution strategy for assigning elements to CPUs using a
```

Comment:
I'm not sure I really have a change request here for this PR, but it's not very encouraging to me that none of this is directly applicable to DG-FD hydro simulations. Those have quite a bit of load imbalance from switching between different schemes. The challenge in that case is we don't know if we need to do FD until _after_ we've initialized the element. This would mean more of a load balance than an initial element distribution. I think that being able to actually balance those simulations is extremely high priority. We can merge this PR without that, but it won't help the majority of our current science applications at all, unfortunately. Nils Vu's requests for some way of specifying a per-element weight is necessary in that case.